src/core/constants.ts

```diff
@@ -180,3 +186,26 @@ export const STORAGE_CACHE_TTL = 30 * 24 * 60 * 60 * 1000;
 export const PLAYER_BAR_SELECTOR: string = "ytmusic-player-bar";
 export const AD_PLAYING_ATTR: string = "is-advertisement";
 export const LYRICS_AD_OVERLAY_ID: string = "blyrics-ad-overlay";
+
+export type SyncType = "syllable" | "word" | "line" | "unsynced";
+
+export interface ProviderConfig {
+  key: string;
+  displayName: string;
+  syncType: SyncType;
+  priority: number;
+}
+
+export const PROVIDER_CONFIGS: ProviderConfig[] = [
+  { key: "bLyrics-richsynced", displayName: "Better Lyrics", syncType: "syllable", priority: 0 },
+  { key: "musixmatch-richsync", displayName: "Musixmatch", syncType: "word", priority: 1 },
+  { key: "yt-captions", displayName: "Youtube Captions", syncType: "line", priority: 2 },
+  { key: "bLyrics-synced", displayName: "Better Lyrics", syncType: "line", priority: 3 },
+  { key: "lrclib-synced", displayName: "LRCLib", syncType: "line", priority: 4 },
+  { key: "legato-synced", displayName: "Legato", syncType: "line", priority: 5 },
+  { key: "musixmatch-synced", displayName: "Musixmatch", syncType: "line", priority: 6 },
+  { key: "yt-lyrics", displayName: "Youtube", syncType: "unsynced", priority: 7 },
+  { key: "lrclib-plain", displayName: "LRCLib", syncType: "unsynced", priority: 8 },
+];
+
+export const LYRIC_SOURCE_KEYS = PROVIDER_CONFIGS.map(p => p.key) as unknown as readonly string[];
```

Comment:
```suggestion
export const LYRIC_SOURCE_KEYS = PROVIDER_CONFIGS.map(p => p.key);
```

Type `LyricSourceKey[]` is inferred!